src/gui/src/Widgets/Rotary.cpp

```diff
@@ -20,324 +20,247 @@
-	float delta = 1.0;
 
-	// Control Modifier = fine adjustment
-	if (ev->modifiers() == Qt::ControlModifier) {
-		stepfactor = 1.0;
-	}
-	if ( !m_bUseIntSteps ) {
-		float fRange = m_fMax - m_fMin;
-		delta = fRange / 100.0;
-	}
-	if ( ev->angleDelta().y() < 0 ) {
-		delta *= -1.;
+	if ( m_type != Type::Small ) {
+		QRectF arcRect( 9.951, 2.2, 24.5, 24.5 );
+
+		if ( m_type == Type::Normal ) {
+			int nStartAngle = 210 * 16; // given in 1/16 of a degree
+			int nSpanAngle  = static_cast<int>( -239 * 16 * ( m_fValue - m_fMin ) / ( m_fMax - m_fMin ) );
+
+			painter.setPen( QPen( colorArc, 1.7 ) );
+			painter.drawArc( arcRect, nStartAngle, nSpanAngle );
+		} else {
+			// Type::Center
+
+			// There will be a special indication of the
+			// center. Either as a gray dot or a bigger green one if
+			// the value is smaller than +/-1% of the range around 0.
+			if ( std::fabs( m_fValue - 0.5 * ( m_fMax + m_fMin ) ) < 0.01 * ( m_fMax - m_fMin ) ) {
+				
+				painter.setPen( QPen( colorArcCenterSet, 2.5 ) );
+				painter.drawArc( arcRect, 90 * 16, -3 * 16 );
+				
+			} else {
+				
+				painter.setPen( QPen( colorArcCenterUnset, 2.5 ) );
+				painter.drawArc( arcRect, 90 * 16, -3 * 16 );
+
+				int nStartAngle = -18 * 16;
+				int nSpanAngle  = static_cast<int>( -200* 16 * ( m_fValue - 0.5 * ( m_fMax + m_fMin ) ) / ( m_fMax - m_fMin ) );
+				if ( m_fValue - 0.5 * ( m_fMax + m_fMin ) < 0 ) {
+					nStartAngle *= -1;
+					nStartAngle -= 2 * 16;
+				}
+				nStartAngle += 89 * 16;
+				
+				painter.setPen( QPen( colorArc, 1.7 ) );
+				painter.drawArc( arcRect, nStartAngle, nSpanAngle ); 
+			}
+		}
 	}
-	setValue( getValue() + (delta * stepfactor) );
-	emit valueChanged(this);
 	
-	if ( m_bShowValueToolTip ) {
-		char tmp[20];
-		sprintf( tmp, "%#.2f", m_fValue );
-		
-		// Problem: ValueToolTip remains visible after wheel event
-		//m_pValueToolTip->showTip( mapToGlobal( QPoint( -38, 1 ) ), QString( tmp ) );
-		
-		// useful but graphically incoeherent
-		QToolTip::showText( ev->globalPos(), QString( "%1" ).arg( m_fValue, 0, 'f', 2 ) , this );
-	}
-}
-
-
-
-void Rotary::mouseMoveEvent( QMouseEvent *ev )
-{
-	if ( m_bIgnoreMouseMove ) {
-		return;
-	}
-
-	float fRange = m_fMax - m_fMin;
-
-	float deltaY = ev->y() - m_fMousePressY;
-	float fNewValue = ( m_fMousePressValue - ( deltaY / 100.0 * fRange ) );
-
-	setValue( fNewValue );
-	emit valueChanged(this);
-
-	if ( m_bShowValueToolTip ) {
-		char tmp[20];
-		sprintf( tmp, "%#.2f", m_fValue );
-		m_pValueToolTip->showTip( mapToGlobal( QPoint( -38, 1 ) ), QString( tmp ) );
-	}
-}
-
-
-
-void Rotary::setMin( float fMin )
-{
-	m_fMin = fMin;
-	update();
-}
-
-float Rotary::getMin()
-{
-	return m_fMin;
-}
-
-
-
-void Rotary::setMax( float fMax )
-{
-	m_fMax = fMax;
-	update();
-}
-
-
-float Rotary::getMax()
-{
-	return m_fMax;
-}
-
-
-void Rotary::setDefaultValue( float fDefaultValue )
-{
-	if ( fDefaultValue == m_fDefaultValue ) {
-		return;
+	float fPi = std::acos( -1 );
```

Comment:
Why aren't you using `M_PI` from `math.h`?

Reply:
I've read on Stackoverflow that `M_PI` is not defined in the C++ standard itself and platform specific.